x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty, flatten } from 'lodash';
+import { callClusterFactory } from '../../../xpack_main';
+import { getCollectorLogger, sendBulkPayload, monitoringBulk } from './lib';
+
+export class BulkUploader {
+  constructor(server, collectorSet, { interval, combineTypes }) {
+    this._timer = null;
+
+    if (typeof interval !== 'number') {
+      throw new Error('interval number of milliseconds is required');
+    }
+    if (typeof combineTypes !== 'function') {
+      throw new Error('combineTypes function is required');
+    }
+
+    this._interval = interval;
+    this._combineTypes = combineTypes;
+    this._log = getCollectorLogger(server);
+
+    this._client = server.plugins.elasticsearch.getCluster('admin').createClient({
+      plugins: [monitoringBulk],
+    });
+
+    this._callClusterWithInternalUser = callClusterFactory(server).getCallClusterInternal();
+
+    /*
+     * Start the interval
+     */
+    this.start = () => {
```

Comment:
The only reason is because I don't want this class to have a `this.collectorSet`, which it would need if `start()` wasn't able to reach the collectorSet via enclosure. It's not a big deal, just my personal preference.

What do you think about that reason?

Reply:
I'm fine with the reasoning. Perhaps add a comment to note that's why?